pkg/occlient/occlient.go

```diff
@@ -984,9 +985,19 @@ func uniqueAppendOrOverwriteEnvVars(existingEnvs []corev1.EnvVar, envVars ...cor
 		mapExistingEnvs[newEnvVar.Name] = newEnvVar
 	}
 
-	// Convert map to slice
-	for _, envVar := range mapExistingEnvs {
-		retVal = append(retVal, envVar)
+	// append the values to the final slice
+	// don't loop because we need them in order
+	for _, envVar := range existingEnvs {
+		if val, ok := mapExistingEnvs[envVar.Name]; ok {
+			retVal = append(retVal, val)
+			delete(mapExistingEnvs, envVar.Name)
```

Comment:
would it be a better idea to just sort them instead of doing this?

Reply:
The order of the env variables is important if you start sorting, you will change the order.
This function is called `appendOrOvewrite` it shouldn't do any side effects like sorting